app/src/main/java/com/keylesspalace/tusky/util/MediaUtils.java

```diff
@@ -88,30 +88,51 @@ public static long getMediaSize(@NonNull ContentResolver contentResolver, @Nulla
     }
 
     @Nullable
-    public static Bitmap getImageThumbnail(ContentResolver contentResolver, Uri uri,
-                                            @Px int thumbnailSize) {
+    public static Bitmap getSampledBitmap(ContentResolver contentResolver, Uri uri, @Px int reqWidth, @Px int reqHeight) {
+        // First decode with inJustDecodeBounds=true to check dimensions
+        final BitmapFactory.Options options = new BitmapFactory.Options();
+        options.inJustDecodeBounds = true;
         InputStream stream;
         try {
             stream = contentResolver.openInputStream(uri);
         } catch (FileNotFoundException e) {
+            Log.w(TAG, e);
             return null;
         }
-        Bitmap source = BitmapFactory.decodeStream(stream);
-        if (source == null) {
-            IOUtils.closeQuietly(stream);
+
+        BitmapFactory.decodeStream(stream, null, options);
+
+        IOUtils.closeQuietly(stream);
+
+        // Calculate inSampleSize
+        options.inSampleSize = calculateInSampleSize(options, reqWidth, reqHeight);
+
+        // Decode bitmap with inSampleSize set
+        options.inJustDecodeBounds = false;
+        try {
+            stream = contentResolver.openInputStream(uri);
+            return BitmapFactory.decodeStream(stream, null, options);
+        } catch (FileNotFoundException e) {
+            Log.w(TAG, e);
+            return null;
+        } catch (OutOfMemoryError e) {
```

Comment:
As far as I know, trying to catch OOM is pointless because nothing else can be done

Reply:
I am kinda torn on this one
[Here](https://stackoverflow.com/questions/2679330/catching-java-lang-outofmemoryerror) people say it's only for cleanup/
[But here](https://stackoverflow.com/questions/7138645/catching-outofmemoryerror-in-decoding-bitmap#7138785) people say it's okay when decoding Bitmap
So I guess this is fine?

Reply:
Yes its fine, I use this in multiple apps. And I tested it with my slowest phone and an 70 megapixel photo xD

Thx for reviewing!